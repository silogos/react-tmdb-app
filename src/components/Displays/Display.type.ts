export type DisplayErrorProps = {
  onReload?: () => void;
};
