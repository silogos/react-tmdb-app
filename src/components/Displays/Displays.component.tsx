import { InformationCircleIcon } from "@heroicons/react/24/outline";
import { DisplayErrorProps } from "./Display.type";

export function DisplayNoData() {
  return (
    <div className="flex flex-row justify-center items-center rounded-lg bg-black bg-opacity-50 p-8 text-gray-300">
      <InformationCircleIcon className="size-8 mr-2" />
      Data Not Found
    </div>
  );
}

export function DisplayError({ onReload }: DisplayErrorProps) {
  return (
    <div className="flex flex-col justify-center items-center rounded-lg bg-red-400 bg-opacity-50 p-8 text-gray-300">
      <div className="flex flex-row items-center gap-2 text-center">
        <InformationCircleIcon className="size-8" />
        Something went wrong
      </div>
      {onReload && (
        <button
          onClick={onReload}
          className="mt-4 py-2 px-4 bg-red-300 text-gray-900 rounded-lg"
        >
          Reload
        </button>
      )}
    </div>
  );
}
