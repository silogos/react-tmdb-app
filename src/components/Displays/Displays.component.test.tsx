import { describe, it, expect, vitest } from "vitest";
import { DisplayNoData, DisplayError } from "./Displays.component";
import { fireEvent, render } from "@testing-library/react";

describe("Display Component", () => {
  it("renders DisplayNoData correctly", () => {
    const component = render(<DisplayNoData />);

    expect(component.getByText("Data Not Found")).toBeInTheDocument();
    expect(component.container).toMatchSnapshot();
  });

  describe("DisplayError", () => {
    it("renders correctly", () => {
      const component = render(<DisplayError onReload={() => {}} />);

      expect(component.getByText("Something went wrong")).toBeInTheDocument();
      expect(component.getByText("Reload")).toBeInTheDocument();
    });

    it("renders correctly when onReload undefined", () => {
      const component = render(<DisplayError />);

      expect(component.queryByText(/Reload/i)).not.toBeInTheDocument();
      expect(component.container).toMatchSnapshot();
    });

    it("renders correctly when onReload undefined", () => {
      const onReload = vitest.fn();
      const component = render(<DisplayError onReload={onReload} />);

      fireEvent.click(component.getByText(/Reload/i));

      expect(onReload).toHaveBeenCalledTimes(1);
      expect(component.container).toMatchSnapshot();
    });
  });
});
