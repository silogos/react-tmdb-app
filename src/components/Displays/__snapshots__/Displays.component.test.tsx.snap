// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Display Component > DisplayError > renders correctly when onReload undefined 1`] = `
<div>
  <div
    class="flex flex-col justify-center items-center rounded-lg bg-red-400 bg-opacity-50 p-8 text-gray-300"
  >
    <div
      class="flex flex-row items-center gap-2 text-center"
    >
      <svg
        aria-hidden="true"
        class="size-8"
        data-slot="icon"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m11.25 11.25.041-.02a.75.75 0 0 1 1.063.852l-.708 2.836a.75.75 0 0 0 1.063.853l.041-.021M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Zm-9-3.75h.008v.008H12V8.25Z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      Something went wrong
    </div>
  </div>
</div>
`;

exports[`Display Component > DisplayError > renders correctly when onReload undefined 2`] = `
<div>
  <div
    class="flex flex-col justify-center items-center rounded-lg bg-red-400 bg-opacity-50 p-8 text-gray-300"
  >
    <div
      class="flex flex-row items-center gap-2 text-center"
    >
      <svg
        aria-hidden="true"
        class="size-8"
        data-slot="icon"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m11.25 11.25.041-.02a.75.75 0 0 1 1.063.852l-.708 2.836a.75.75 0 0 0 1.063.853l.041-.021M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Zm-9-3.75h.008v.008H12V8.25Z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      Something went wrong
    </div>
    <button
      class="mt-4 py-2 px-4 bg-red-300 text-gray-900 rounded-lg"
    >
      Reload
    </button>
  </div>
</div>
`;

exports[`Display Component > renders DisplayNoData correctly 1`] = `
<div>
  <div
    class="flex flex-row justify-center items-center rounded-lg bg-black bg-opacity-50 p-8 text-gray-300"
  >
    <svg
      aria-hidden="true"
      class="size-8 mr-2"
      data-slot="icon"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m11.25 11.25.041-.02a.75.75 0 0 1 1.063.852l-.708 2.836a.75.75 0 0 0 1.063.853l.041-.021M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Zm-9-3.75h.008v.008H12V8.25Z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
    Data Not Found
  </div>
</div>
`;
